evaluation/tasks/wmt/wmt.py

```diff
@@ -1,2 +1,55 @@
 # Module for any additional processing required for the WMT dataset
 # HuggingFace dataset link: https://huggingface.co/datasets/wmt19
+import torch
+from datasets import load_dataset
+from torch.utils.data import DataLoader, Dataset
+from tqdm import tqdm
+
+from evaluation.tasks.auto_task import AutoTask
+
+
+class WMTEnglishDataset(Dataset):
+    def __init__(self, tokenizer, stride=512, max_len=1024, pair="kk-en"):
+        super().__init__()
+        assert "en" in pair, f"Expected `pair` to contain English, but got {pair} instead"
+        wmt = load_dataset("wmt19", pair, split="validation")["translation"]
+        text_list = [item["en"] for item in wmt]
+        text = " ".join(text_list)
+        input_ids = tokenizer(text, return_tensors="pt", verbose=False).input_ids.squeeze()
+        self.input_ids = input_ids.unfold(size=max_len, step=stride, dimension=-1)
+
+    def __len__(self):
+        return len(self.input_ids)
+
+    def __getitem__(self, index):
+        return self.input_ids[index]
+
+
+class WMTTask(AutoTask):
+    @staticmethod
+    def get_display_name() -> str:
+        return "wmt"
+
+    def evaluate(self) -> None:
+        stride = self.task_config["stride"]
+        dataset = WMTEnglishDataset(
+            self.tokenizer, stride=stride, max_len=self.model.config.n_positions, pair=self.task_config["pair"]
+        )
+        loader = DataLoader(
+            dataset,
+            batch_size=self.task_config["batch_size"],
+            num_workers=self.task_config["num_workers"],
+            shuffle=False,
+            drop_last=True,
+        )
+        log_likelihoods = []
+        for input_ids in tqdm(loader, desc=f"Evaluating {self.get_display_name()}"):
+            input_ids = input_ids.to(self.device)
+            target_ids = input_ids.clone()
+            target_ids[:, :-stride] = -100
```

Comment:
nit: The code looks good. But for the sake of readability (in a public repo), I typically like to add some simple comments explaining what and why is being done in operations like these.

Reply:
Thanks for the feedback! Just added a comment line above to indicate that it's for ignoring context tokens during loss computation. Hope this is fine!